src/compiler/proposals-and-stabilization/custom-test-jobs.md

```diff
@@ -0,0 +1,96 @@
+# Adding custom test jobs or adding custom test components to rust-lang/rust CI
+
+## Scope
+
+This policy is applicable to proposals for adding new test jobs or adjusting existing test jobs to build additional custom things that are *not* shipped to the end users, but may still cause [rust-lang/rust] PR CI or Full CI (the "CI") to fail, or produce a failure message, which may impact compiler/library/tools contributors who also use the [rust-lang/rust] CI.
+
+For example, this includes but is not limited to:
+
+- Ecosystem test jobs, such as Rust for Linux or Fuchsia;
+- Custom codegen backend test components, such as `codegen_gcc`, which are not part of the "main" distributed toolchain.
+
+## Background
+
+[rust-lang/rust] runs a small set of test (and build) jobs on PR CI (faster, less expensive/exhaustive), and runs a much larger set of test (and build) jobs on Full CI. PR CI usually takes around an hour, while Full CI usually takes around 3 hours. Having test jobs that:
+
+- may spuriously fail; or
+- may genuinely fail but it's not clear who should be consulted about the failure or who is responsible for fixing the failure; or
+- otherwise do not have well documented test job maintainers and failure protocol;
+
+can introduce a lot of friction and frustration for other contributors who now will have to deal with the failure or understand why the test job failed in the first place, which may now be blocking their PR. Everyone utilizing the [rust-lang/rust] CI is expected to use it responsibly.
+
+To help with this, please follow the process described below if you'd like to propose adding a custom test job/component to the [rust-lang/rust] CI, and copy and fill out the Custom Test Job/Component Policy.
```

Comment:
Problem (for myself): I need to explicitly note here (as described in the MCP) that this policy is intended to be **retroactive**.

Reply:
I'm not sure that the policy needs to say it is retrospective, as that has no bearing on anyone reading this later to propose another job. We can just implement this policy for previously added jobs.

Reply:
That is true, that only really makes sense for the MCP.